### PR TITLE
Fix ambiguities about the Kafka CDC connector

### LINKED ISSUE #396: Ambiguity: what is CDC?
### Where's the bug?

https://www.risingwave.dev/docs/latest/create-source-cdc

### Describe the bug

In this page, you mentioned a concept called `CDC`. I wonder what CDC is, and how I can use it? What is `Materialized CDC`? What is `Debezium`? It seems that to use CDC, we also need to install Kafka, correct? Any documentation on that?

Some references:
* Flink CDC code: https://github.com/ververica/flink-cdc-connectors
* Flink CDC doc: https://ververica.github.io/flink-cdc-connectors/master/
* Kafka CDC ingestion: https://developer.confluent.io/learn-kafka/data-pipelines/kafka-data-ingestion-with-cdc/

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Add information about CDC in general, materialized and non-materialized sources, and the tools needed to convert database changes to Kafka topics. 

The revamp is still in progress, and may be delayed as there are some issues with the connector node. Therefore, I'm adding the same changes to this topic.
- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves #396 

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/create-source/create-source-cdc.md [hunk 1]
```diff
@@ -5,15 +5,21 @@ description: Connect RisingWave to a CDC source.
...
+Change Data Capture (CDC) refers to the process of identifying and capturing data changes in a database, then delivering the changes to a downstream service in real time. 
 
-To subscribe to a Kafka topic that reads data from a CDC tool, use the `CREATE SOURCE` command to create a source connector.
+CDC tools and platforms can record row-level changes (INSERT, UPDATE, and DELETE activities) that apply to tables in a upstream database and stream the data change event records to event streaming platforms such as Kafka. You can connect RisingWave to the Kafka topics to receive the data changes.
 
-:::note
+To ingest CDC data from MySQL or PostgreSQL into RisingWave, you can use a CDC tool to convert data change streams in databases to Kafka topics, and then use the native Kafka connector in RisingWave to consume data from the Kafka topics.
 
-Currently, RisingWave only supports materialized CDC sources with primary keys, and the data format must be Debezium JSON or Maxwell JSON.
+For RisingWave to ingest CDC data, you must create a materialized source connection (`CREATE MATERIALIZED SOURCE`) and specify primary keys.
```
Comment (tabVersion) [inline docs/create-source/create-source-cdc.md]: I don't think source is a connection. suggest removing `connection` here.

**Review by tabVersion — APPROVED** [review]
great work!

File: docs/create-source/create-source-cdc.md [hunk 2]
```diff
@@ -5,15 +5,21 @@ description: Connect RisingWave to a CDC source.
...
+Change Data Capture (CDC) refers to the process of identifying and capturing data changes in a database, then delivering the changes to a downstream service in real time. 
 
-To subscribe to a Kafka topic that reads data from a CDC tool, use the `CREATE SOURCE` command to create a source connector.
+CDC tools and platforms can record row-level changes (INSERT, UPDATE, and DELETE activities) that apply to tables in a upstream database and stream the data change event records to event streaming platforms such as Kafka. You can connect RisingWave to the Kafka topics to receive the data changes.
 
-:::note
+To ingest CDC data from MySQL or PostgreSQL into RisingWave, you can use a CDC tool to convert data change streams in databases to Kafka topics, and then use the native Kafka connector in RisingWave to consume data from the Kafka topics.
```
Comment (neverchanje) [inline docs/create-source/create-source-cdc.md:12]: We allow built-in CDC  without the need of using Kafka in our future versions. Probably we should leave a notice here.

File: docs/create-source/create-source-cdc.md (hunk 1 above)
Comment (neverchanje) [reply docs/create-source/create-source-cdc.md]: Agree

File: docs/create-source/create-source-cdc.md (hunk 2 above)
Comment (tabVersion) [reply docs/create-source/create-source-cdc.md:12]: connector node is not ready to publish. Letting user continues with kafka connect still meets expectation.

Comment (hengm3467) [reply docs/create-source/create-source-cdc.md:12]: @neverchanje @tabVersion Thanks for the clarification!

File: docs/create-source/create-source-cdc.md [hunk 3]
```diff
@@ -5,15 +5,21 @@ description: Connect RisingWave to a CDC source.
 slug: /create-source-cdc
 ---
 
-Change Data Capture (CDC) refers to the process of identifying and capturing data changes in a database, then delivering the changes to a downstream service in real time. <br/> CDC tools and platforms can record row-level changes (INSERT, UPDATE, and DELETE activities) that apply to tables in a upstream database and stream the data change event records to event streaming platforms such as Kafka. You can connect RisingWave to the Kafka topics to receive the data changes.
+Change Data Capture (CDC) refers to the process of identifying and capturing data changes in a database, then delivering the changes to a downstream service in real time. 
 
-To subscribe to a Kafka topic that reads data from a CDC tool, use the `CREATE SOURCE` command to create a source connector.
+CDC tools and platforms can record row-level changes (INSERT, UPDATE, and DELETE activities) that apply to tables in a upstream database and stream the data change event records to event streaming platforms such as Kafka. You can connect RisingWave to the Kafka topics to receive the data changes.
```
Comment (CharlieSYH) [inline docs/create-source/create-source-cdc.md]: a upstream -> an upstream

File: docs/create-source/create-source-cdc.md [hunk 4]
```diff
@@ -5,15 +5,21 @@ description: Connect RisingWave to a CDC source.
...
 
-:::note
+To ingest CDC data from MySQL or PostgreSQL into RisingWave, you can use a CDC tool to convert data change streams in databases to Kafka topics, and then use the native Kafka connector in RisingWave to consume data from the Kafka topics.
 
-Currently, RisingWave only supports materialized CDC sources with primary keys, and the data format must be Debezium JSON or Maxwell JSON.
+For RisingWave to ingest CDC data, you must create a materialized source connection (`CREATE MATERIALIZED SOURCE`) and specify primary keys.
+
+The difference between a non-materialized and materialized source connection is data from a materialized source connection is stored in RisingWave, while data from a non-materialized source connection is not.
```
Comment (CharlieSYH) [inline docs/create-source/create-source-cdc.md]: The difference between a non-materialized and materialized source connection is 
->
The difference between a non-materialized and materialized source connection is _that_

File: docs/create-source/create-source-cdc.md [hunk 5]
```diff
@@ -5,15 +5,21 @@ description: Connect RisingWave to a CDC source.
...
+
+The supported CDC data formats are [Debezium](https://debezium.io) JSON (for both MySQL and PostgreSQL) and [Maxwell](https://maxwells-daemon.io) JSON (for MySQL only). 
+
+- Debezium JSON (`ROW FORMAT DEBEZIUM_JSON`): You can use the [Debezium connector for MySQL](https://debezium.io/documentation/reference/stable/connectors/mysql.html) to convert MySQL or PostgreSQL data change streams to Kafka topics. To learn about how to configure MySQL and deploy the Debezium connector for MySQL, see the [Debezium connector for MySQL documentation](https://debezium.io/documentation/reference/stable/connectors/mysql.html).
+- Maxwell JSON (`ROW FORMAT MAXWELL`): You can use [Maxwell's daemon](https://maxwells-daemon.io/) to convert MySQL data changes to kafka topics. To learn about how to configure MySQL and deploy Maxwell's daemon, see the [Quick Start](https://maxwells-daemon.io/quickstart/).
```
Comment (CharlieSYH) [inline docs/create-source/create-source-cdc.md]: kafka topics -> Kafka topics

**Review by CharlieSYH — APPROVED** [review]

**Pull request merged**